presto-elasticsearch/src/main/java/io/prestosql/elasticsearch/ElasticsearchClient.java

```diff
@@ -258,26 +300,32 @@ private static void validateCertificates(KeyStore keyStore)
         }
     }
 
-    public Map<String, Node> getNodes()
+    private Set<Node> fetchNodes()
     {
         NodesResponse nodesResponse = doRequest("_nodes/http", NODES_RESPONSE_CODEC::fromJson);
 
-        ImmutableMap.Builder<String, Node> result = ImmutableMap.builder();
+        ImmutableSet.Builder<Node> result = ImmutableSet.builder();
         for (Map.Entry<String, NodesResponse.Node> entry : nodesResponse.getNodes().entrySet()) {
             String nodeId = entry.getKey();
             NodesResponse.Node node = entry.getValue();
 
             if (node.getRoles().contains("data")) {
-                result.put(nodeId, new Node(nodeId, node.getHttp().getAddress()));
+                result.add(new Node(nodeId, node.getHttp().getAddress()));
             }
         }
 
         return result.build();
     }
 
+    public Set<Node> getNodes()
+    {
+        return nodes.get();
+    }
+
     public List<Shard> getSearchShards(String index)
     {
-        Map<String, Node> nodeById = getNodes();
+        Map<String, Node> nodeById = getNodes().stream()
```

Comment:
any reason to not index in the background loader?

Reply:
Not really, other than I want the public method to return a set of nodes (to support the system table) instead of a Map of id->node.